### PR TITLE
Introduce reference client and server

### PR DESCRIPTION
**Pull request by siggy**
Add two new commands, `ref-client` and `ref-server`, to perform a single
bi-directional gRPC request, to aide in evaluating HTTP/2 and gRPC proxy
behavior.

Signed-off-by: Andrew Seigner <siggy@buoyant.io>

### CODE REVIEW

File: refclient/refclient.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}(stream, receiving, count)
+
+	err = stream.Send(&pb.StreamingResponseSpec{
+		Count: int32(count),
+	})
+	if err != nil {
+		log.Errorf("stream.Send returned %s", err)
+	}
+
+	<-receiving
```
Comment (adleong) [inline refclient/refclient.go]: One thing that's a bit weird here is that we delay closing the client stream until after we have finished receiving the server stream.  Do we do this specifically to trigger issues in the go-grpc library?  A more typical thing to do would be to close the client stream immediately after we're done sending.  i.e. immediately after stream.Send

File: refclient/refclient.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	defer conn.Close()
+	client := pb.NewResponderClient(conn)
+
+	stream, err := client.StreamingGet(context.Background())
+	if err != nil {
+		log.Fatalf("%v.StreamingGet(_) = _, %v", client, err)
+	}
+
+	count := -1
```
Comment (adleong) [inline refclient/refclient.go]: This invalid input is designed to trigger failures on the server?  In other words, this reference client is more like a unit test, testing a boundry case?

File: refclient/refclient.go (hunk 1 above)
Comment (siggy) [reply refclient/refclient.go]: That's correct, we're waiting for the server to return an error before calling `CloseSend` specifically to trigger the go-grpc issue, where `CloseSend` never sends an EOS.

File: refclient/refclient.go (hunk 2 above)
Comment (siggy) [reply refclient/refclient.go]: That's correct, I just wanted the server to return immediately. A more complete implementation could take the count as a command-line param.

Comment (siggy) [reply refclient/refclient.go]: Yup, my first instinct was to use a separate repo, but it’s quite convenient having the protobuf already in place for this repo.

**Pull request merged**